### PR TITLE
Database V01 Changes, Regions & Districts

### PR DESCRIPTION
**Pull request by AdvSol-Darrel**

### CODE REVIEW

**Comment by DSoLetsDev** [comment]
Looks good! Confirmed the right number of entities were present according to the diagrams.

File: api/Crt.Data/Repositories/DistrictRepository.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+{
+    public interface IDistrictRepository
+    {
+        IEnumerable<DistrictDto> GetAllDistricts();
+        Task<IEnumerable<DistrictDto>> GetAllDistrictsAsync();
+        Task<DistrictDto> GetDistrictByDistrictId(decimal id);
+        Task<DistrictDto> GetDistrictByDistrictNumber(decimal number);
+    }
+
+    public class DistrictRepository : CrtRepositoryBase<CrtDistrict>, IDistrictRepository
```
Comment (ychung-mot) [inline api/Crt.Data/Repositories/DistrictRepository.cs:19]: Need to get only active ones using end date.

File: api/Crt.Data/Repositories/RegionRepository.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+{
+    public interface IRegionRepository
+    {
+        IEnumerable<RegionDto> GetAllRegions();
+        Task<IEnumerable<RegionDto>> GetAllRegionsAsync();
+        Task<RegionDto> GetRegionByRegionNumber(decimal regionNumber);
+        Task<RegionDto> GetRegionByRegionId(decimal id);
+    }
+
+    public class RegionRepository : CrtRepositoryBase<CrtRegion>, IRegionRepository
```
Comment (ychung-mot) [inline api/Crt.Data/Repositories/RegionRepository.cs:19]: Need to get only active ones using end date.

File: api/Crt.Domain/Services/DistrictService.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+namespace Crt.Domain.Services
+{
+    public interface IDistrictService
+    {
+        IEnumerable<DistrictDto> GetAllDistricts();
+        Task<IEnumerable<DistrictDto>> GetAllDistrictsAsync();
+        Task<DistrictDto> GetDistrictByDistrictId(decimal id);
+    }
+
+    public class DistrictService : IDistrictService
```
Comment (ychung-mot) [inline api/Crt.Domain/Services/DistrictService.cs:18]: Need to return only active ones.

**Review by ychung-mot — CHANGES_REQUESTED** [review]

**Pull request merged**